target/linux/mpc85xx/p2020/config-default

```diff
@@ -0,0 +1,31 @@
+CONFIG_ARCH_HAS_TICK_BROADCAST=y
+CONFIG_CPU_RMAP=y
+CONFIG_GENERIC_CLOCKEVENTS_BROADCAST=y
+CONFIG_GENERIC_IRQ_MIGRATION=y
+CONFIG_GENERIC_TBSYNC=y
+CONFIG_HAVE_RCU_TABLE_FREE=y
+CONFIG_LOCK_SPIN_ON_OWNER=y
+#CONFIG_MMC=y
```

Comment:
# CONFIG_MMC is not set
# CONFIG_MMC_SDHCI is not set
?

Reply:
Sorry. Dumb mistake.

Reply:
@chunkeey 
> Do you know what's wrong with the SD-Card reader though? Is it a chip on the pcie/usb bus or is it just missing the driver?

P2020 processor have dedicated peripherial. At this moment SD-Card not working. Freescale kernel tree (kernel 4.1) have patches, which wasn't sent to/ rejected in upstream. Also in Turris project, sd-reader was used. 
 I plan to work with this after work with VSC7385.

Reply:
It looks like parts of the driver are upstream.

I found a [fsl,p2020-esdhc and fsl,p2020-rev1-esdhc binding](https://github.com/torvalds/linux/blob/master/Documentation/devicetree/bindings/mmc/fsl-esdhc.txt) that is serviced by a
driver under /drivers/mmc host/sdhci-of-esdhc.c that just needs
CONFIG_MMC_SDHCI_PLTFM and CONFIG_MMC_SDHCI_OF_ESDHC.

So I wonder why it isn't working out of the box? But ok, you can of course add it at any time you see fit.

Reply:
> [    1.170204] mmc0: Internal clock never stabilised.
> [    1.207150] mmc0: Internal clock never stabilised.

These messages are interesting. Could you check in /proc/device-tree if you can find the esdhc node there and see if the "clock-frequency" property file has any meaningful content (i.e. not null - hexdump can help here).

Reply:
@chunkeey 
Clock have value != 0:

```
root@OpenWrt:/# hexdump /proc/device-tree/soc@ffe00000/sdhc@2e000/clock-frequenc
y
0000000 0ee6 b280
```

In u-boot everything work. I can read data from reader and check info about card.